File: S/1281-subtract-the-product-and-sum-of-digits-of-an-integer/main.go
package main

import (
	"fmt"
)

func subtractProductAndSum(n int) int {
	add, multi := 0, 1
	for n/10 > 0 {
		m := n % 10
		add += m
		multi *= m
		n /= 10
	}
	m := n % 10
	add += m
	multi *= m
	return multi - add
}

func main() {
	fmt.Println(subtractProductAndSum(234))
	fmt.Println(subtractProductAndSum(4420))
}

File: S/1281-subtract-the-product-and-sum-of-digits-of-an-integer/main.rs
impl Solution {
  pub fn subtract_product_and_sum(n: i32) -> i32 {
    let mut digits: Vec<i32> = Vec::new();
    let mut n = n;
    while n > 0 {
      digits.push(n % 10);
      n /= 10;
    }
    digits.clone().into_iter().product::<i32>() - digits.clone().into_iter().sum::<i32>()
  }
}
